rktl_game/nodes/game_manager

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+"""Use ball location to determine when a goal is scored.
+License:
+  BSD 3-Clause License
+  Copyright (c) 2022, Autonomous Robotics Club of Purdue (Purdue ARC)
+  All rights reserved.
+"""
+
+
+
+# ROS
+import rospy
+from std_msgs.msg import Bool, Header
+from nav_msgs.msg import Odometry
+from rktl_msgs.msg import Score, MatchStatus
+from std_srvs.srv import Empty, EmptyResponse
+
+class ScoreKeeper():
+    def __init__(self):
+        rospy.init_node('score_keeper')
+
+        self.orange_score = 0
+        self.blue_score = 0
+        self.orange_goal = (rospy.get_param('/field/length', 1) - 0.15) / 2 
+        self.blue_goal = (rospy.get_param('/field/length', 1)  - 0.15) / -2
+        self.enabled = False
+        self.match_status = 0
+        self.game_clock = rospy.get_param('/game_length', 90)
+        self.manager_rate = rospy.get_param('manager_rate', 10)
+
+        # Publishers
+        self.match_status_pub = rospy.Publisher('match_status', MatchStatus, queue_size=1)
+        self.active_pub = rospy.Publisher('cars/enable', Bool, queue_size=1)
+
+        # Subscribers
+        rospy.Subscriber('ball/odom', Odometry, self.check_goal)
+        rospy.Subscriber('cars/enable', Bool, self.enable)
+
+        # Services
+        rospy.Service('reset_game', Empty, self.reset)
+        rospy.Service('unpause_game', Empty, self.unpause)
+        rospy.Service('pause_game', Empty, self.pause)
+
+        # Counts loops and decrements game clock every second
+        self.timer_ctr = 0
+        # main loop
+        rate = rospy.Rate(self.manager_rate)
+        while not rospy.is_shutdown():
+            if self.match_status == 1:
+                self.timer_ctr += 1
+                if self.timer_ctr == self.manager_rate:
+                    self.timer_ctr = 0
+                    self.game_clock -= 1
+            self.update_status()
+            rate.sleep()
+
+    def update_status(self):
+        if (self.match_status == 1 and self.game_clock == 0):
+            if (self.orange_score > self.blue_score):
+                self.match_status = 3
+            else:
+                self.match_status = 4
+        score = Score(self.orange_score, self.blue_score)
+        header = Header()
+        header.stamp = rospy.Time.now()
+        self.match_status_pub.publish(MatchStatus(header, self.match_status, self.game_clock, score))
+
+    def pause(self, _):
+        self.enabled = False
+        self.match_status = 0
+        self.active_pub.publish(False)
+        return EmptyResponse()
+
+    def reset(self, _):
+        self.orange_score = 0
+        self.blue_score = 0
+        self.enabled = False
+        self.match_status = 0
+        self.game_clock = rospy.get_param('/game_length', 90)
```

Comment:
Rather than fetching this param twice, and having duplicated code and default values, I'd save it to a member variable like `self.GAME_LENGTH`. Or maybe fetch it without the default value in both locations to enforce that it needs to be in a config file.

Reply:
It's fetched when you reset in case it as been modified by the webUI. It's not the prettiest solution, but it works for now